rate_limiter.py

```
@@ -1,12 +1,14 @@
 import time
 import threading
 from collections import defaultdict
+from typing import Tuple, Optional
```

Ruff (0.11.9)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)